### ISSUE TITLE
[Bug]: 1.20 Update Creates a CTD from the JDT Inventory Holder and Supplimentaries

### ISSUE THREAD

**Issue by zeziba** [AllTheMods/ATM-10#1161]
### Possible Fixes

Yes

### Modpack Version

1.20

### What happened?

After upgrading to 1.20 and loading my base once I move towards a certain location of the base it will CTD with the cause as the game client not having enough memory or so it reports on crash. 

From looking at the game log file it seems that Supplementaries is causing a collision with Just Dire Thing's Inventory Holder's in world display of the player model.

My specific use case of this particular item is I have 8 set-up in a row where I have different levels of gear I have created held within. The in-world display of character model allows for the cosmetic's to override the model as well which is what I have currently.

Each of the inventory holders are out of render distance and as I walk towards them the game will with 100% certainty crash as this is the 4th time I have tested this particular issue.

```

[16Nov2024 00:18:30.433] [Render thread/ERROR] [net.minecraft.client.Minecraft/FATAL]: Reported exception thrown!
net.minecraft.ReportedException: Rendering entity in world
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.client.renderer.blockentity.BlockEntityRenderDispatcher.tryRender(BlockEntityRenderDispatcher.java:111) ~[client-1.21.1-20240808.144430-srg.jar%23564!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.client.renderer.blockentity.BlockEntityRenderDispatcher.render(BlockEntityRenderDispatcher.java:79) ~[client-1.21.1-20240808.144430-srg.jar%23564!/:?]
	at TRANSFORMER/sodium@0.6.0-beta.4+mc1.21.1/net.caffeinemc.mods.sodium.client.render.SodiumWorldRenderer.renderBlockEntity(SodiumWorldRenderer.java:418) ~[sodium-neoforge-0.6.0-beta.4+mc1.21.1.jar%23885!/:?]
	at TRANSFORMER/sodium@0.6.0-beta.4+mc1.21.1/net.caffeinemc.mods.sodium.client.render.SodiumWorldRenderer.renderBlockEntities(SodiumWorldRenderer.java:353) ~[sodium-neoforge-0.6.0-beta.4+mc1.21.1.jar%23885!/:?]
	at TRANSFORMER/sodium@0.6.0-beta.4+mc1.21.1/net.caffeinemc.mods.sodium.client.render.SodiumWorldRenderer.renderBlockEntities(SodiumWorldRenderer.java:314) ~[sodium-neoforge-0.6.0-beta.4+mc1.21.1.jar%23885!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.client.renderer.LevelRenderer.handler$bjc000$sodium$onRenderBlockEntities(LevelRenderer.java:11895) ~[client-1.21.1-20240808.144430-srg.jar%23564!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.client.renderer.LevelRenderer.renderLevel(LevelRenderer.java:1081) ~[client-1.21.1-20240808.144430-srg.jar%23564!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.client.renderer.GameRenderer.renderLevel(GameRenderer.java:1271) ~[client-1.21.1-20240808.144430-srg.jar%23564!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.client.renderer.GameRenderer.render(GameRenderer.java:1024) ~[client-1.21.1-20240808.144430-srg.jar%23564!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.client.Minecraft.runTick(Minecraft.java:1195) ~[client-1.21.1-20240808.144430-srg.jar%23564!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.client.Minecraft.run(Minecraft.java:807) ~[client-1.21.1-20240808.144430-srg.jar%23564!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.client.main.Main.main(Main.java:230) ~[client-1.21.1-20240808.144430-srg.jar%23564!/:?]
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103) ~[?:?]
	at java.base/java.lang.reflect.Method.invoke(Method.java:580) ~[?:?]
	at MC-BOOTSTRAP/fml_loader@4.0.31/net.neoforged.fml.loading.targets.CommonLaunchHandler.runTarget(CommonLaunchHandler.java:136) ~[loader-4.0.31.jar%23101!/:4.0]
	at MC-BOOTSTRAP/fml_loader@4.0.31/net.neoforged.fml.loading.targets.CommonLaunchHandler.clientService(CommonLaunchHandler.java:124) ~[loader-4.0.31.jar%23101!/:4.0]
	at MC-BOOTSTRAP/fml_loader@4.0.31/net.neoforged.fml.loading.targets.CommonClientLaunchHandler.runService(CommonClientLaunchHandler.java:32) ~[loader-4.0.31.jar%23101!/:4.0]
	at MC-BOOTSTRAP/fml_loader@4.0.31/net.neoforged.fml.loading.targets.CommonLaunchHandler.lambda$launchService$4(CommonLaunchHandler.java:118) ~[loader-4.0.31.jar%23101!/:4.0]
	at MC-BOOTSTRAP/cpw.mods.modlauncher@11.0.4/cpw.mods.modlauncher.LaunchServiceHandlerDecorator.launch(LaunchServiceHandlerDecorator.java:30) [modlauncher-11.0.4.jar%23104!/:?]
	at MC-BOOTSTRAP/cpw.mods.modlauncher@11.0.4/cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:53) [modlauncher-11.0.4.jar%23104!/:?]
	at MC-BOOTSTRAP/cpw.mods.modlauncher@11.0.4/cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:71) [modlauncher-11.0.4.jar%23104!/:?]
	at MC-BOOTSTRAP/cpw.mods.modlauncher@11.0.4/cpw.mods.modlauncher.Launcher.run(Launcher.java:103) [modlauncher-11.0.4.jar%23104!/:?]
	at MC-BOOTSTRAP/cpw.mods.modlauncher@11.0.4/cpw.mods.modlauncher.Launcher.main(Launcher.java:74) [modlauncher-11.0.4.jar%23104!/:?]
	at MC-BOOTSTRAP/cpw.mods.modlauncher@11.0.4/cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:26) [modlauncher-11.0.4.jar%23104!/:?]
	at MC-BOOTSTRAP/cpw.mods.modlauncher@11.0.4/cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:23) [modlauncher-11.0.4.jar%23104!/:?]
	at cpw.mods.bootstraplauncher@2.0.2/cpw.mods.bootstraplauncher.BootstrapLauncher.run(BootstrapLauncher.java:210) [bootstraplauncher-2.0.2.jar:?]
	at cpw.mods.bootstraplauncher@2.0.2/cpw.mods.bootstraplauncher.BootstrapLauncher.main(BootstrapLauncher.java:69) [bootstraplauncher-2.0.2.jar:?]
Caused by: java.lang.ClassCastException: class com.direwolf20.justdirethings.client.blockentityrenders.InventoryHolderBER$1 cannot be cast to class net.mehvahdjukaar.supplementaries.api.IQuiverEntity (com.direwolf20.justdirethings.client.blockentityrenders.InventoryHolderBER$1 is in module justdirethings@1.4.9 of loader 'TRANSFORMER' @5950054d; net.mehvahdjukaar.supplementaries.api.IQuiverEntity is in module supplementaries@1.21-3.0.13-alpha of loader 'TRANSFORMER' @5950054d)
	at TRANSFORMER/supplementaries@1.21-3.0.13-alpha/net.mehvahdjukaar.supplementaries.client.renderers.entities.layers.QuiverLayer.render(QuiverLayer.java:43) ~[supplementaries-1.21-3.0.15-beta-neoforge.jar%23901!/:?]
	at TRANSFORMER/supplementaries@1.21-3.0.13-alpha/net.mehvahdjukaar.supplementaries.client.renderers.entities.layers.QuiverLayer.render(QuiverLayer.java:24) ~[supplementaries-1.21-3.0.15-beta-neoforge.jar%23901!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.client.renderer.entity.LivingEntityRenderer.render(LivingEntityRenderer.java:133) ~[client-1.21.1-20240808.144430-srg.jar%23564!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.client.renderer.entity.player.PlayerRenderer.render(PlayerRenderer.java:72) ~[client-1.21.1-20240808.144430-srg.jar%23564!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.client.renderer.entity.player.PlayerRenderer.render(PlayerRenderer.java:46) ~[client-1.21.1-20240808.144430-srg.jar%23564!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.client.renderer.entity.EntityRenderDispatcher.render(EntityRenderDispatcher.java:159) ~[client-1.21.1-20240808.144430-srg.jar%23564!/:?]
	at TRANSFORMER/justdirethings@1.4.9/com.direwolf20.justdirethings.client.blockentityrenders.InventoryHolderBER.renderMockPlayerEntity(InventoryHolderBER.java:95) ~[justdirethings-1.4.9.jar%23748!/:?]
	at TRANSFORMER/justdirethings@1.4.9/com.direwolf20.justdirethings.client.blockentityrenders.InventoryHolderBER.render(InventoryHolderBER.java:36) ~[justdirethings-1.4.9.jar%23748!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.client.renderer.blockentity.BlockEntityRenderDispatcher.setupAndRender(BlockEntityRenderDispatcher.java:90) ~[client-1.21.1-20240808.144430-srg.jar%23564!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.client.renderer.blockentity.BlockEntityRenderDispatcher.lambda$render$0(BlockEntityRenderDispatcher.java:79) ~[client-1.21.1-20240808.144430-srg.jar%23564!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.client.renderer.blockentity.BlockEntityRenderDispatcher.tryRender(BlockEntityRenderDispatcher.java:104) ~[client-1.21.1-20240808.144430-srg.jar%23564!/:?]
	... 26 more
[16Nov2024 00:18:31.411] [Server thread/INFO] [net.minecraft.server.MinecraftServer/]: Stopping server
[16Nov2024 00:18:31.414] [Server thread/INFO] [net.minecraft.server.MinecraftServer/]: Saving players
[16Nov2024 00:18:31.437] [Server thread/INFO] [net.minecraft.server.network.ServerGamePacketListenerImpl/]: zeziba lost connection: Disconnected
[16Nov2024 00:18:31.437] [Server thread/INFO] [net.minecraft.server.MinecraftServer/]: zeziba left the game
[16Nov2024 00:18:31.465] [Server thread/INFO] [net.minecraft.server.network.ServerCommonPacketListenerImpl/]: Stopping singleplayer server as player logged out
[16Nov2024 00:18:31.465] [Server thread/INFO] [net.minecraft.server.MinecraftServer/]: Saving worlds
[16Nov2024 00:18:31.587] [Server thread/INFO] [QuarryPlus/QuarryChunkLoader]: Remove chunk loading at x=1047, z=152
```

### latest.log

https://gist.github.com/zeziba/3ecf6cec28318428b3be575700f13a57

**Comment by DaddyLongspoon** [comment]
me too.  my world is bricked until this is sorted :( I have five inventory holders, world crashes immediately.  I haven't even looked at supplementaries

**Comment by ImMorphi** [comment]
+1

**Comment by alekthefirst** [comment]
See https://github.com/Direwolf20-MC/JustDireThings/issues/283
Fixed in JDT 1.5.0, released with ATM10 2.0
